File: arf/GeneralNode.cpp
/*************************************************************************
                           GeneralNode  -  description
                             -------------------
    début                : 22/01/2016
    copyright            : (C) 2016 par jfolleas
*************************************************************************/

//------------------ class <GeneralNode> (file GeneralNode.cpp) -----------------

//---------------------------------------------------------------- INCLUDE

//-------------------------------------------------------- Include système
using namespace std;
#include <iostream>
#include <vector>

//------------------------------------------------------ Include personnel
#include "GeneralNode.h"

 GeneralNode::~GeneralNode()
{}


GeneralNode* GeneralNode::getFather()
{return this->father;}

bool GeneralNode::SonIsLeaf(int place)
{return false;}

bool GeneralNode::GetValue()
{return true;}

void GeneralNode::Split(const int leafSplited, const int leafTrue)
{}

void GeneralNode::SetValue()
{}

void GeneralNode::ChangeLeafValue(const int & place , const bool & value)
{}

void GeneralNode::changeChildValue(GeneralNode* const & ,const bool &)
{}

File: arf_python/GeneralNode.py
# -------- Aim of the file

# This file provide an abstract class with method to dicscretise a point

# -------- Import

from abc import ABCMeta, abstractmethod

# --------- Constant


# --------- Code
class GeneralNode:
    """
    This class is the interface for all data discretisators.
    Args :
    :param lambda_error: float representing how each coordinates of the points should be ceil or floor.
    """
    __metaclass__ = ABCMeta

    def __init__(self, father = None):
        self.father = father

    def get_father (self):
        """
        return the father of the leaf.
        :return:
        """
        return self.father

    @abstractmethod
    def get_value(self):
        """
        return the value of the leaf.
        """
        raise NotImplementedError

    @abstractmethod
    def set_value (self):
        """
        Return a list of points dicretize to insert in the Bloom filter.
        """
        raise NotImplementedError

    @abstractmethod
    def change_child_value(self, place, hasElement ):
        """
        Change the value of a child
        """
        raise NotImplementedError

    @abstractmethod
    def change_leaf_value(self, general_node, hasElement):
        """
        Change the value of a leaf.
        """
        raise NotImplementedError

    @abstractmethod
    def split(self, place_leaf_splitted):
        """
        The leaf become a node.
        """
        raise NotImplementedError

    @abstractmethod
    def son_is_leaf(self, place):
        """
        Teste if the son of a node is a leaf.
        """
        raise NotImplementedError

    @abstractmethod
    def print_data(place):
        """
        Teste if the son of a node is a leaf.
        """
        raise NotImplementedError